File: ios/RNDrawView.m
//
//  RNDrawView.m
//  RNViewRecorder
//
//  Created by Toddle on 13/12/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(RNDrawView, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(brushWidth, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(colorString, NSString)
RCT_EXPORT_VIEW_PROPERTY(isErasing, BOOL)
RCT_EXPORT_VIEW_PROPERTY(imageBackgroundUri, NSString)
RCT_EXPORT_VIEW_PROPERTY(drawingTool, NSString)
RCT_EXPORT_VIEW_PROPERTY(pointPath, NSString)
RCT_EXPORT_VIEW_PROPERTY(opacity, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(forceOnTap, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(forceSensitive, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(pointStep, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(coreProportion, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(colorAlpha, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(brushes, NSArray)



RCT_EXTERN_METHOD(resetCanvas:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(undo:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(redo:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(saveAsImage:(nonnull NSNumber *)node)


RCT_EXPORT_VIEW_PROPERTY(onEventStackUpdated, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onViewMount, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onImageStored, RCTDirectEventBlock)

//onViewMount

@end

File: ios/RNDrawView.swift
@objc(RNDrawView)
class RNDrawView: RCTViewManager {
    override func view() -> DrawingCanvas! {
        let drawView = DrawingCanvas();
        
        return drawView;
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    
    @objc func resetCanvas(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! DrawingCanvas                                      // 5
            component.clear()                           // 6
        }
    }
    
    @objc func undo(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! DrawingCanvas                                       // 5
            component.undo()                           // 6
        }
    }
    
    @objc func redo(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! DrawingCanvas                                     // 5
            component.redo()                           // 6
        }
    }
    
    //saveAsImage
    @objc func saveAsImage(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! DrawingCanvas                                       // 5
            component.saveAsImage()                           // 6
        }
    }
    
   
    
}
